src/H5private.h

```diff
@@ -401,23 +401,19 @@
 #endif /* H5_HAVE_PARALLEL */
 
 /*
- * Types and max sizes for POSIX I/O.
- * OS X (Darwin) is odd since the max I/O size does not match the types.
+ * Types and max I/O size for POSIX read/write calls
  */
 #if defined(H5_HAVE_WIN32_API)
-#define h5_posix_io_t         unsigned int
-#define h5_posix_io_ret_t     int
-#define H5_POSIX_MAX_IO_BYTES INT_MAX
-#elif defined(H5_HAVE_DARWIN)
-#define h5_posix_io_t         size_t
-#define h5_posix_io_ret_t     ssize_t
-#define H5_POSIX_MAX_IO_BYTES INT_MAX
+#define h5_posix_io_t     unsigned int
+#define h5_posix_io_ret_t int
 #else
-#define h5_posix_io_t         size_t
-#define h5_posix_io_ret_t     ssize_t
-#define H5_POSIX_MAX_IO_BYTES SSIZE_MAX
+#define h5_posix_io_t     size_t
+#define h5_posix_io_ret_t ssize_t
```

Comment:
There's no longer a need to handle Darwin/MacOS separately, so we just need to handle Window's weird types.

Reply:
A bit of a "all the world's Linux" choice, but probably not terrible.

Reply:
More of a "lowest common denominator"